### PR TITLE
Project skeleton

### PR DESCRIPTION
**Pull request by macrologist**
A first draft of a project skeleton generator. I will be using it in the comming days on a few systems, so it is subject to change. 

Suggestions and complaints welcome!

### CODE REVIEW

**Comment by svetlyak40wt** [comment]
It would be cool to see [Mystic](https://github.com/roswell/mystic) template.

**Comment by macrologist** [comment]
@svetlyak40wt I might consider doing that with good enough reason. For me, it comes down to whether or not the extra power is worth introducing all those dependencies. For this particular case, I'm not sure that the power is needed -- my crappy ad-hoc template only uses a couple of variables.

If you can help me see the light on using Mystic, I'm amenable.

**Comment by svetlyak40wt** [comment]
@macrologist ok!

Just watch my video about Mystic: https://www.youtube.com/watch?v=lwcmOZ0DQf0

Turn on english subtitles!

File: src/project/project.lisp [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+   (format-template-expand
+    (a:read-file-into-string template)
+    args)
+   expanded))
+
+(defun join (strings sep)
+  "join a list of strings together."
+  (with-output-to-string (out)
+    (loop :for (str . more) :on strings
+          :do (princ str out)
```
Comment (stylewarning) [inline src/project/project.lisp:87]: i prefer WRITE-STRING

File: src/project/project.lisp [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    :do (setf str (replace-all
+                   str
+                   (format nil "{{~a}}" var)
+                   val))
+    :finally (return str)))
+
+(defun walk-directory-tree (dir fn)
+  "Apply FN to each NON-DIRECTORY file under filesystem tree rooted at
+the directory DIR."
+ (dolist (file (uiop:directory-files dir))
```
Comment (stylewarning) [inline src/project/project.lisp:125]: (map nil fn ...)

File: template/script.lisp [hunk 3]
```diff
@@ -0,0 +1,9 @@
+(require '#:asdf)
+
+(asdf:load-system '#:{{LIBRARY}})
```
Comment (stylewarning) [inline template/script.lisp:3]: it's now preferred to use lowercase string names for ASDF system designators I believe

**Review by stylewarning — APPROVED** [review]
little things. is this good to merge?

**Comment by macrologist** [comment]
> little things. is this good to merge?

Oh, I guess so. Its good enough to use right now but is a fairly brittle system. If somebody actually requests further features it will need to be scrapped and rethought - which is fine IMO.

**Pull request merged**